### PR TITLE
virttest.libvirt_vm: Add support for virtqueues while guest boot

### PR DESCRIPTION
**Pull request by sathnaga**
Let's add support for virt-queues during guest boot
in virt-install commandline, queues and net_driver
params to be set accordingly in-order to use this support.

Signed-off-by: Satheesh Rajendran <sathnaga@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by sathnaga** [comment]
Covered all possible combinations of multivm ang multi nic with different queues and driver settings:

```
avocado run boot --vt-type libvirt --vt-guest-os RHEL.7.7.ppc64le \
--vt-extra-params take_regular_screendumps=no \
backup_image_before_testing=no \
virtinstall_qemu_cmdline=' -M pseries,ic-mode=xics' \
libvirt_controller=virtio-scsi scsi_hba=virtio-scsi-pci drive_format=scsi-hd \
create_vm_libvirt=yes env_cleanup=yes kill_vm=yes kill_vm_libvirt=yes \
os_variant=rhel7.0 \
vms="vm1 vm2" main_vm=vm1 master_images_clone=img1 \
nics="nic1 nic2" \
queues_nic1=100 queues_nic2_vm2=50 \
net_driver_nic1=vhost net_driver_nic2_vm2=qemu \
flexible_nic_index=yes
JOB ID     : 34fe5ddc9a3ea9dfc1b013cca072b921506570d9
JOB LOG    : /home/sath/tests/results/job-2019-07-17T06.01-34fe5dd/job.log
 (1/1) io-github-autotest-qemu.boot: PASS (532.10 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 533.94 s
JOB HTML   : /home/sath/tests/results/job-2019-07-17T06.01-34fe5dd/results.html
```


```
2019-07-17 06:02:02,981 libvirt_vm       L1990 INFO |     --name 'vm1'
2019-07-17 06:02:02,981 libvirt_vm       L1990 INFO |     --machine pseries
2019-07-17 06:02:02,981 libvirt_vm       L1990 INFO |     --memory=1024
2019-07-17 06:02:02,981 libvirt_vm       L1990 INFO |     --vcpu=2
2019-07-17 06:02:02,981 libvirt_vm       L1990 INFO |     --import
2019-07-17 06:02:02,982 libvirt_vm       L1990 INFO |     --vnc
2019-07-17 06:02:02,982 libvirt_vm       L1990 INFO |     --os-variant rhel7.0
2019-07-17 06:02:02,982 libvirt_vm       L1990 INFO |     --serial pty
2019-07-17 06:02:02,982 libvirt_vm       L1990 INFO |     --memballoon model=virtio
2019-07-17 06:02:02,982 libvirt_vm       L1990 INFO |     --controller type=scsi,model=virtio-scsi
2019-07-17 06:02:02,982 libvirt_vm       L1990 INFO |     --disk path=/home/sath/tests/data/avocado-vt/images/rhel77-ppc64le_vm1.qcow2,bus=scsi,size=10,format=qcow2
2019-07-17 06:02:02,982 libvirt_vm       L1990 INFO |     --network=bridge=virbr0,model=virtio,driver_queues=100,driver_name=vhost,mac=52:54:00:a5:a6:a7
2019-07-17 06:02:02,982 libvirt_vm       L1990 INFO |     --network=bridge=virbr0,model=virtio,driver_queues=1,mac=52:54:00:a8:a9:aa
2019-07-17 06:02:02,982 libvirt_vm       L1990 INFO |     --noautoconsole


2019-07-17 06:02:06,044 libvirt_vm       L1990 INFO |     --name 'vm2'
2019-07-17 06:02:06,044 libvirt_vm       L1990 INFO |     --machine pseries
2019-07-17 06:02:06,044 libvirt_vm       L1990 INFO |     --memory=1024
2019-07-17 06:02:06,044 libvirt_vm       L1990 INFO |     --vcpu=2
2019-07-17 06:02:06,044 libvirt_vm       L1990 INFO |     --import
2019-07-17 06:02:06,044 libvirt_vm       L1990 INFO |     --vnc
2019-07-17 06:02:06,044 libvirt_vm       L1990 INFO |     --os-variant rhel7.0
2019-07-17 06:02:06,044 libvirt_vm       L1990 INFO |     --serial pty
2019-07-17 06:02:06,045 libvirt_vm       L1990 INFO |     --memballoon model=virtio
2019-07-17 06:02:06,045 libvirt_vm       L1990 INFO |     --controller type=scsi,model=virtio-scsi
2019-07-17 06:02:06,045 libvirt_vm       L1990 INFO |     --disk path=/home/sath/tests/data/avocado-vt/images/rhel77-ppc64le_vm2.qcow2,bus=scsi,size=10,format=qcow2
2019-07-17 06:02:06,045 libvirt_vm       L1990 INFO |     --network=bridge=virbr0,model=virtio,driver_queues=100,driver_name=vhost,mac=52:54:00:ab:ac:ad
2019-07-17 06:02:06,045 libvirt_vm       L1990 INFO |     --network=bridge=virbr0,model=virtio,driver_queues=50,driver_name=qemu,mac=52:54:00:ae:af:b0
2019-07-17 06:02:06,045 libvirt_vm       L1990 INFO |     --noautoconsole
```

File: virttest/utils_net.py [hunk 1]
```diff
@@ -2448,7 +2448,7 @@ class VirtIface(propcan.PropCan, object):
     """
 
     __slots__ = ['nic_name', 'g_nic_name', 'mac', 'nic_model', 'ip',
-                 'nettype', 'netdst']
+                 'nettype', 'netdst', 'queues', 'net_driver']
```
Comment (balamuruhans) [inline virttest/utils_net.py]: please make this change as separate commit.

File: virttest/libvirt_vm.py [hunk 2]
```diff
@@ -875,8 +877,13 @@ def add_nic(help_text, nic_params):
                     result += ",model=%s" % nic_model
                 if nettype and mac:
                     result += ',mac=%s' % mac
+                if nettype and nic_queues and has_sub_option('network', 'driver_queues'):
+                    result += ',driver_queues=%s' % nic_queues
+                    if nic_driver and has_sub_option('network', 'driver_name'):
+                        result += ',driver_name=%s' % nic_driver
                 elif mac:  # possible to specify --mac w/o --network
                     result += " --mac=%s" % mac
+
```
Comment (balamuruhans) [inline virttest/libvirt_vm.py]: this line can be removed.

File: shared/cfg/base.cfg [hunk 3]
```diff
@@ -73,6 +73,13 @@ netdst = virbr0
 #advanced network configuration of the host
 #host_ip_addr = ""
 
+# For enabling virt queues in tests uncomment below two params and
+# modify based on need, net_driver value can either be vhost or qemu
+# vhost - use host kernel, qemu - userspace, bydefault it is vhost if
+# kernel supports
+# queues = 10
```
Comment (balamuruhans) [inline shared/cfg/base.cfg]: will the param name `virt_queues` be better ?

Comment (sathnaga) [reply shared/cfg/base.cfg]: @balamuruhans we are reusing the param that is used by qemu tests aswell, so this should be fine for now, will work on a different PR later to change the name.

Comment (balamuruhans) [reply shared/cfg/base.cfg]: okay

**Comment by sathnaga** [comment]
@balamuruhans addressed.
